### PR TITLE
[REF] stock_account: valuation layer and not admin user

### PR DESCRIPTION
**Pull request by sle-odoo**
Carefully sudo when necessary.

task-2034044

### CODE REVIEW

File: addons/stock_account/models/product.py [hunk 1]
```diff
@@ -235,7 +236,7 @@ def _change_standard_price(self, new_price, counterpart_account_id=False):
         account_moves.post()
 
         # Actually update the standard price.
-        self.write({'standard_price': new_price})
+        self.with_context(force_company=company_id.id).sudo().write({'standard_price': new_price})
```
Comment (amoyaux) [inline addons/stock_account/models/product.py]: force_company is probably useless since sudo ref

File: addons/stock_account/models/stock_move.py [hunk 2]
```diff
@@ -262,7 +262,7 @@ def _action_done(self, cancel_backorder=False):
         # For every in move, run the vacuum for the linked product.
         products_to_vacuum = valued_moves['in'].mapped('product_id')
         for product_to_vacuum in products_to_vacuum:
-            product_to_vacuum._run_fifo_vacuum()
+            product_to_vacuum._run_fifo_vacuum(self.env.company)
```
Comment (amoyaux) [inline addons/stock_account/models/stock_move.py]: It would by default use self.env.company

File: addons/stock_account/models/product.py (hunk 1 above)
Comment (sle-odoo) [reply addons/stock_account/models/product.py]: i'm not sure. it's still used almost everywhere (find pricelist, in ir.property, _select_seller,..)

**Comment by sle-odoo** [comment]
robodoo r+

**Pull request closed without merging**